ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ClusterTerminator.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import io.confluent.ksql.util.ExecutorUtil;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ClusterTerminator {
+
+  private final KsqlConfig ksqlConfig;
+  private final KsqlEngine ksqlEngine;
+
+  public ClusterTerminator(
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine
+  ) {
+    Objects.requireNonNull(ksqlConfig, "ksqlConfig is null.");
+    Objects.requireNonNull(ksqlEngine, "ksqlEngine is null.");
+    this.ksqlConfig = ksqlConfig;
+    this.ksqlEngine = ksqlEngine;
+  }
+
+  @SuppressWarnings("unchecked")
+  public void terminateCluster(final List<String> deleteTopicPatterns) {
+    validateDeleteList(deleteTopicPatterns);
+    terminatePersistentQueries();
+    deleteSinkTopics(deleteTopicPatterns);
+    deleteCommandTopic();
+    ksqlEngine.close();
+  }
+
+  private void terminatePersistentQueries() {
+    new ArrayList<>(ksqlEngine.getPersistentQueries()).stream()
+        .map(PersistentQueryMetadata::getQueryId)
+        .forEach(queryId -> ksqlEngine.terminateQuery(queryId, true));
+  }
+
+  private void validateDeleteList(final List<String> deleteTopicPatterns) {
+    if (deleteTopicPatterns.isEmpty()) {
+      return;
+    }
+    ksqlEngine.getMetaStore().getAllKsqlTopics().values().stream()
+        .forEach(ksqlTopic -> {
+          deleteTopicPatterns.forEach(
+              deleteTopicPattern -> {
+                if (ksqlTopic.getKafkaTopicName().equals(deleteTopicPattern)
```

Comment:
This will have edge cases.  Consider, if I pass the _pattern_ `mytopic.`. This should match any topic that starts with `mytopic` and is followed by any other single char, e.g. `mytopic1`.

The check you're doing would compare the _pattern_ against the name of the topic. So if there was a non-sink topic called `mytopic.` you'd throw an exception.

Now, personally, I don't think we need this check at all. (See comment above). However, if you want to have the check, then you should make sure it works correctly. So you need more tests to cover such things.

To implement properly you'd need to remove any non-literals from `deleteTopicPatterns` before running your check.  I *think* you can detect a literal by using `Pattern.quote`, which will espace any special characters, so it follows that:

```java
final boolean isExact = Pattern.quote(somePattern).equals(somePattern);
```

i.e. if the quoted version is the same as the original, then its a pattern that is looking for an exact match and has no special regex characters in it.   You need good testing around something like this to ensure there are no edge cases, e.g. what happens if `somePattern` contains an escaped regex special character, like `mytopic\.`?  (I'm fairly sure the `quote` approach will handle this)